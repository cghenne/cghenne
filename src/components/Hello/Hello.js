// @flow
import React from 'react';
import './Hello.css';

const Hello = () => {
  return (
    <div className="box hello">
      <h3>Hello</h3>
      <p>
        I am a dynamic and capable front end engineer with many years of
        experience in development, maintenance and support of applications and
        websites. My preferred framework is ReactJS. In my free time, I am
        currently learning about functional programming and Clojure.
      </p>
    </div>
  );
};

Hello.displayName = 'Hello';

export default Hello;
