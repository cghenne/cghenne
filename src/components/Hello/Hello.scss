@import '../../styles/utils.scss';
.hello {
  grid-area: hello;

  @include mq($from: tablet) {
    h3 {
      text-align: right;
    }
  }

  @media print {
    h3 {
      text-align: right;
    }
  }
}
