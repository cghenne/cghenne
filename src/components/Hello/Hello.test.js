import React from 'react';
import { shallow } from 'enzyme';
import Hello from './Hello';

describe('Hello', () => {
  const wrapper = shallow(<Hello />);

  it('should render without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render title', () => {
    expect(wrapper.find('h3').text()).toEqual('Hello');
  });

  it('should render an introduction paragraph', () => {
    expect(wrapper.find('p').exists()).toEqual(true);
  });

  it('should render the box with a class of box and hello', () => {
    expect(wrapper.find('div').prop('className')).toEqual('box hello');
  });
});
