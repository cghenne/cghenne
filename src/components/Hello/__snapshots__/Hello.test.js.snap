// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hello should render without crashing 1`] = `
<div
  className="box hello"
>
  <h3>
    Hello
  </h3>
  <p>
    I am a dynamic and capable JavaScript engineer with multiple years of experience in development and maintenance of applications. I also have experience hiring, training and mentoring teams. My favourite framework is React.
  </p>
</div>
`;
